lib/rules/no-assignment-of-untracked-properties-used-in-tracking-contexts.js

```diff
@@ -74,29 +158,64 @@ module.exports = {
       return {};
     }
 
+    const options = context.options[0] || { extraMacros: [] };
+
+    const macroConfigurations = [...DEFAULT_MACRO_CONFIGURATIONS, ...options.extraMacros];
+
+    const macrosByPath = new Map();
+    const macrosByIndexPath = new Map();
+
+    macroConfigurations.forEach((config) => {
+      if (!macrosByPath.has(config.path)) {
+        macrosByPath.set(config.path, new Map());
+      }
+      macrosByPath.get(config.path).set(config.name, config);
+
+      if (!macrosByIndexPath.has(config.indexPath)) {
+        macrosByIndexPath.set(config.indexPath, new Map());
+      }
+      if (!macrosByIndexPath.get(config.indexPath).has(config.indexName)) {
+        macrosByIndexPath.get(config.indexPath).set(config.indexName, new Map());
+      }
+      macrosByIndexPath.get(config.indexPath).get(config.indexName).set(config.name, config);
+    });
+
     // State being tracked for this file.
-    let computedImportName = undefined;
     let trackedImportName = undefined;
+    let computedImportName = undefined;
     let setImportName = undefined;
-    let macroImportNames = new Map();
+    let macrosByName = new Map();
+    let macrosByIndexName = new Map();
 
     // State being tracked for the current class we're inside.
     const classStack = new Stack();
 
     return {
       ImportDeclaration(node) {
+        const importPath = node.source.value;
+
+        if (macrosByPath.has(importPath)) {
+          const newMacroImportNameEntries = [...macrosByPath.get(importPath)]
+            .map(([name, config]) => [getImportIdentifier(node, importPath, name), config])
+            .filter(([importedName]) => importedName);
```

Comment:
If this is supposed to filter out empty ones, you could use `Boolean(importedName)`?

Reply:
Wrapping the results of a `filter` callback in `Boolean` doesn't do anything, since `filter` checks whether the returned value is truthy. It would be redundant.

Reply:
Yep, just thought it might clarify the intent. Feel free to disregard.